File: src/Models/API/UserInfo.cs
using System;
using ApplicationChallenge.Models.Database;

namespace ApplicationChallenge.Models.API
{
    public class UserInfo
    {
        public string AccountId { get; set; }
        public string Nickname { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string LinkedInUrl { get; set; }

        public ApplicantInfo Applicant { get; set; }
        public CompanyInfo Company { get; set; }

        public class ApplicantInfo
        {
            public Skill[] Skills { get; set; }
            public bool Available { get; set; }
            public string Biography { get; set; }
        }

        public class CompanyInfo
        {
            public string Name { get; set; }
            public string Biography { get; set; }
        }
    }
}

File: src/ClientApp/src/app/models/user-info.ts
import {Skill} from "./skill.model";

/**
 * @description Publicly accessible user info. Does not contain any private details such as email
 */
export class UserInfo {
  accountId: string;
  nickname: string;
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  linkedInUrl: string;

  applicant: { skills: Skill[], available: boolean, biography: string };
  company: { name: string, biography: string };
}
